### ISSUE TITLE
ImgDrive 1.6.5 beta

### ISSUE THREAD

**Issue by imgdrive** [dvdforge/imgdrive_translations#35]
http://download.yubsoft.com/ImgDrive_1.6.5.exe

ImgDrive 1.6 has a new installer which written in C, smaller, simpler and faster!

Change log:
- New installer with more setup options
- Added: "Tools -> Number of Drives -> Hide Unmounted Drives" menu item
- Added: change drive name
- Add drive state images: Green-mounted Gray-unmounted Orange-Ejected
- Fixed crash if there is empty TITLE/PERFORMER value in .cue files
- Fixed mount failed in some cases
- Support mount a folder as a drive
- Update menu icons (#29)
- Fixed no prompt message if destination file is already exist while Copying Disc (#30)
- Change "Create ISO Image" dialog layout (#31)
- Add File/Options menu [portable]
- Add "Hide Unmounted Drives" [portable]
- Add "Stay on Top" [portable]
- Add shortcuts [portable]

Version 1.5.8 2018-07-11
- Added: command line support [portable]
- Show a wait dialog while installing driver [portable]
- Only one instance is allowed to run [portable]
- Decrease the program file size [portable]
- Fix a silly mistake when click 'Browse' menu item [portable]
- Update 'Check for Update' feature [portable]
- Fixed sign issue for 64-bit+32-bit combined exe [portable]
- Support drag drop a image file to mount [portable]

**Comment by bovirus** [comment]
@dvdforge 

I'm so sorry but I prefer as usability (for most of the user) the previous installer.
It was more clear and less options to check/enable (good for basic users).

Please make these changes (if you can)

Add more installer languages (ex. Italian)
Add the OS language detection and enable automatically the same language for the installer,.

**Comment by oldman777** [comment]
> I'm so sorry but I prefer as usability (for most of the user) the previous installer.
> It was more clear and less options to check/enable (good for basic users).  

Most of the basic users buy games in "Steam" or somewhere else, and do not use software like that.
And if they use any software, they can press "I agree" and "Install", not paying attention to anything else. 
Sorry for your candor. 

> Add more installer languages (ex. Italian)
> Add the OS language detection and enable automatically the same language for the installer,. 

Totally agree.

**Comment by imgdrive** [comment]
@bovirus 
Thank you for your comment. I will improve this new setup to make it better. 

By default, collapse the tree to make it simpler:
![image](https://user-images.githubusercontent.com/19568093/43652582-2fdda53a-9778-11e8-9393-5eb6cbb9f153.png)

I'll sort out the installer translate text when there are no fatal bugs and huge UI changes.

**Comment by imgdrive** [comment]
When i was learning programming, i found there are many steps/times to install visual studio/MSDN etc, Once time i install Visual Studio 2008 SP1, 10 hours later, it's still installing. What does the installer do, scan my disk to make sure every byte can be access successfully?

After using Mac OSX six years ago, i found most of the Apps don't need installer, just drag and drop, then everything is OK now.  I prefer this way.

**Comment by imgdrive** [comment]
@bovirus @oldman777 
http://download.yubsoft.com/ImgDrive_1.6.2.exe

> Add more installer languages (ex. Italian)

Please translate this file: https://github.com/dvdforge/imgdrive_translations/blob/master/setup_lang/ITA.txt
after committed, i will compile it into .exe.

> Add the OS language detection and enable automatically the same language for the installer,.

Yes, this feature had implemented

and had changed some UI layout, to make it more friendly for users.
![image](https://user-images.githubusercontent.com/19568093/43676753-b61a22a2-9829-11e8-933c-6711082b5b1e.png)
![image](https://user-images.githubusercontent.com/19568093/43676755-bbdd9f16-9829-11e8-9986-c3a4e0b8de4a.png)

**Comment by oldman777** [comment]
Looks great, thank you.

**Comment by bovirus** [comment]
@dvdforge 

Thanks for the version.
ITA.txt translated and commited.

**Comment by oldman777** [comment]
RUS.txt is ready.

**Comment by imgdrive** [comment]
@bovirus @oldman777
Has updated ImgDrive_1.6.2.exe with the latest ITA.txt and rus.txt:
http://download.yubsoft.com/ImgDrive_1.6.2.exe

**Comment by oldman777** [comment]
Perfect on russian! Thank you.

Remain to translate: 

![2018-08-04_191629](https://user-images.githubusercontent.com/28647829/43678392-2f81be0c-981b-11e8-9b39-77b78fa2f20c.jpg)  

![2018-08-04_191654](https://user-images.githubusercontent.com/28647829/43678394-3527cca2-981b-11e8-852e-a3027f38d1ef.jpg)

**Comment by bovirus** [comment]
@dvdforge

I made some small changes to Italian language.

Is it right that no program group is created and no uninstall icon is available?
I prefer to have a program group and an icon to run ImgDrive/uninstall it and visit web site.

I believe that in the final window the string **_%s web site_** should be changed in **_Visit %s web site_** to enhance the meaning of this operation

**Comment by oldman777** [comment]
No string for any language 1.6.2: 

![430480](https://user-images.githubusercontent.com/28647829/43678516-43d6c102-981d-11e8-8ee2-886e4c92414b.jpg)

**Comment by imgdrive** [comment]
@oldman777
> Remain to translate:

new installer has no standalone uninstaller, uninstall ImgDrive will call: imgdrive -xu, add "translate uninstaller" in todo-list. will implement when this new tiny installer has no big design changes.

> Unsupported file format

todo, have plan to refactor mount error source code

**Comment by imgdrive** [comment]
@bovirus 
has changed to Visit %s web site 

had updated ImgDrive_1.6.2.exe with latest ITA.txt:
http://download.yubsoft.com/ImgDrive_1.6.2.exe

> Is it right that no program group is created and no uninstall icon is available?

will add a option to create a Program Group

**Comment by imgdrive** [comment]
> Is it right that no program group is created and no uninstall icon is available?

Have tried to implement this feature, but for a reason then give up:
Windows will give the uninstaller administrator privilege automatically when you click the uninstaller in appwiz.cpl (Control Panel), but not when you click in Start Menu. I think it's impossible to create a shortcut link to main program which will popup UAC dialog when click it, so this is the reason why most of programs should contains a standalone uninstaller.

In short, if the program doesn't have a standalone uninstaller exe, the best way is to avoid create "Uninstall %s" in start menu folder.

I will release ImgDrive with new installer next week, if has some issues caused to add a standalone uninstaller exe,  than i will implement this feature.
